### ISSUE TITLE
Module Federation serving stuck on setup phase

### ISSUE THREAD

**Issue by mnlttt** [nrwl/nx#11174]
## Current Behavior
When using `withModuleFederation` from `@nrwl/angular/module-federation` and the code branches into `traverseUpFileTreeAndPerformActionUntil`, if the library is missing `ng-package.json` (so every lib that isn't buildable?) the serving is stuck at `Generating browser application bundles (phase: setup)...` because it loops endlessy the library folder.

## Expected Behavior
The app should be served.

## Steps to Reproduce
I've migrated the project from Angular 8 to Angular 14. If I expose a newly generated library it works (because the code doesn't branch into that function), but with our already existing non-publishable libraries it doesn't, so something went bad down the road. These existing libraries weren't created with Nx. Help me to find what's causing this problem.

### Environment
```
   Node : 18.4.0
   OS   : darwin x64
   npm  : 8.12.1
   
   nx : 14.4.2
   @nrwl/angular : 14.4.2
   @nrwl/cypress : 14.4.2
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.4.2
   @nrwl/eslint-plugin-nx : 14.4.2
   @nrwl/express : Not Found
   @nrwl/jest : 14.4.2
   @nrwl/js : 14.4.2
   @nrwl/linter : 14.4.2
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.4.2
   @nrwl/web : 14.4.2
   @nrwl/workspace : 14.4.2
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
         @ng-bootstrap/ng-bootstrap: 12.1.2
         ngx-bootstrap: 5.1.1
         rxjs: 6.6.7

```

**Comment by gerardcastell** [comment]
Hi there! My team has faced the same issue, we found out that commenting the module unlocks the septup phase and the compilation succeed, then if we uncomment the module it succeeds! So it seems that there is a problem with the compilation. Any idea about this? thanks!

**Comment by leosvelperez** [comment]
Thanks for reporting this!

Could you please provide a repo where this can be reproduced? I can't reproduce it in a newly created workspace with Module Federation apps and a shared library (non-buildable).

**Comment by mnlttt** [comment]
@leosvelperez I think I found the reason: when you have

```
"@my-org/my-lib/*": [
  "libs/my-lib/src/lib/*"
],
```

and you're barrel exporting, at that point it starts looping because as mentioned in the first post it's missing `ng-package.json`.

Do you think could it be?

**Comment by mnlttt** [comment]
@leosvelperez I just noticed that [aliasesUnderLibrary](https://github.com/nrwl/nx/blob/db3d6cb3a1f3747b4a3582336df7ce7269c8e44a/packages/angular/src/utils/mf/mf-webpack.ts#L55) when in `tsconfig.base.json` I have this:

```
      "@my-org/shared": [
        "libs/shared/src/index.ts"
      ],
      "@my-org/shared/*": [
        "libs/shared/src/lib/*"
      ],
      "@my-org/shared-somethingelse": [
        "libs/shared-somethingelse/src/index.ts"
      ],
```

It get's a different pattern.
Every unique-named lib logs this:

```
[ '@my-org/lib-1/*' ]
[ '@my-org/lib-2/*' ]
[ '@my-org/lib-3/*' ]
```

But if I have two libs named shared and shared-somethingelse I get this:
```
[
  '@my-org/shared/*',
  '@my-org/shared-somethingelse',
  '@my-org/shared-somethingelse/*'
]

And it loops.

**Comment by leosvelperez** [comment]
@mnlttt thanks for the info!

**Comment by ubergeoff** [comment]
I noticed this "freeze up" in the serve/build steps as well.

If I have a lib with the type path of:

```
"@web/stores/sbg-group-store": ["libs/group/stores/sbg-group-store/src/index.ts"],
```

And then I create a custom "testing" lib with the type path of:

```
 "@web/stores/sbg-group-store/testing": ["libs/group/stores/sbg-group-store/mocks/index.ts"],
```

Then the serve/build just freezes.

If I take away this "/testing" path - build works fine again....

**Comment by mnlttt** [comment]
@ubergeoff have you already tried to update nx? For me it has been fixed after #11286

**Comment by ubergeoff** [comment]
Thx @mnlttt  - will give that a try... and confirm later...
